### ISSUE TITLE
Changing Config file properties

### ISSUE THREAD

**Issue by optimaltaskin** [open-mmlab/mmdetection#7790]
Hello,

I have been trying to train an image segmentation neural network with your library. To increase precision I'm using pointrend as well. My present accuracy is generally good but still objects borders are not detected properly. Some images are below:

![card](https://user-images.githubusercontent.com/85254524/164707649-ab8cfecb-c27f-4ba1-9f01-b7da4b46f873.png)
![card2](https://user-images.githubusercontent.com/85254524/164707967-df0e3437-78ef-43d4-9f2a-dda8543ca08e.png)

As can be seen card edges are not straight with those images. Moreover in the first image, mask is restricted with the bounding box. So I have been trying to fine tune config file to increase precision. My current config file is as below:

```
_base_ = '../point_rend/point_rend_r50_caffe_fpn_mstrain_1x_coco_downloaded.py'
model = dict(
    backbone=dict(
        depth=101,
        init_cfg=dict(type='Pretrained',
                      checkpoint='torchvision://resnet101')),
    roi_head=dict(
        bbox_head=dict(
            num_classes=1,
        ),
        mask_head=dict(
            num_classes=1,
            loss_mask=dict(
                type='CrossEntropyLoss', use_mask=True, loss_weight=1.1)),
        point_head=dict(
            num_classes=1,
            loss_point=dict(
                type='CrossEntropyLoss', use_mask=True, loss_weight=1.1))),
    test_cfg=dict(
        rpn=dict(
            nms_pre=1000,
            max_per_img=100,
            nms=dict(type='nms', iou_threshold=0.7),
            min_bbox_size=0),
        rcnn=dict(
            score_thr=0.05,
            nms=dict(type='nms', iou_threshold=0.5),
            max_per_img=10,
            mask_thr_binary=0.5,
            subdivision_steps=5,
            subdivision_num_points=784,
            scale_factor=2))
)

dataset_type = 'COCODataset'
classes = ('card',)
log_config = dict(interval=4)
checkpoint_config = dict(interval=10)
runner = dict(type='EpochBasedRunner', max_epochs=64)

# optimizer = dict(type='Adam', lr=0.0002, weight_decay=0.0001)
optimizer = dict(lr=0.0002, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)

data = dict(
    samples_per_gpu=1,
    workers_per_gpu=1,
    train=dict(
        img_prefix='data/base_set/training/',
        classes=classes,
        ann_file='data/base_set/training/training_labels.json'),
    test=dict(
        img_prefix='data/base_set/testing/',
        classes=classes,
        ann_file='data/base_set/testing/testing_labels.json'),
    val=dict(
        img_prefix='data/base_set/testing/',
        classes=classes,
        ann_file='data/base_set/testing/testing_labels.json'),
    # val=dict(
    #     img_prefix='data/base_set/validation/',
    #     classes=classes,
    #     ann_file='data/base_set/validation/val_labels.json'),
)

load_from = 'checkpoints/point_rend_r50_caffe_fpn_mstrain_1x_coco-1bcb5fb4.pth'

```
So I have two questions:
1) By changing which config properties I can increase precision?
2) When I try to change for instance size of in_channels or out_channels, I always encounter shape mismatch errors. I try to change other layers channels size to match subsequent layers size but always fail. So to change such channel sizes, which document or reference can I check to see which other properties I shall change to prevent size mismatch errors?

Best wishes,

**Comment by BIGWangYuDong** [comment]
You may try to use AdamW as optimizer, which may help you increase the performance. Also, learning rate, batch size, weight decay may increase the precision.

For the second question, you may have a look at the https://github.com/open-mmlab/mmdetection/blob/master/docs/en/tutorials/config.md. in and out channels may also change in other blocks(e.g. if you want to change the in_channel in head, you may also need to change in neck)
